android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt

```diff
@@ -86,15 +89,14 @@ public class HyperTrackPlugin(): FlutterPlugin, MethodCallHandler, StreamHandler
       "isRunning" -> result.success(sdk.isRunning)
       "start" -> start(result, sdk)
       "stop" -> stop(result, sdk)
-      "addGeotag" -> addGeotag(call.arguments(), result, sdk)
+      "addGeotag" -> addGeotag(call.arguments()!!, result, sdk)
       "allowMockLocations" -> allowMockLocations(result, sdk)
-      "setDeviceName" -> setDeviceName(call.arguments(), result, sdk)
-      "setDeviceMetadata" -> setDeviceMetadata(call.arguments(), result, sdk)
+      "setDeviceName" -> setDeviceName(call.arguments()!!, result, sdk)
+      "setDeviceMetadata" -> setDeviceMetadata(call.arguments()!!, result, sdk)
```

Comment:
this will crash if arguments are null

Reply:
The arguments will not be null as Flutter wrapper wont allow that. The method call for these does not accept null parameter

`  void setDeviceMetadata(Map<String, Object> data) => _pluginInterface.setDeviceMetadata(data);`
Object needs to be converted to Object?

Reply:
Okay, then shouldn't the `!!` be removed?

Reply:
Just use ? and raise set result as error if it is null

Reply:
call.arguments()?.let { arguments -> setDeviceName(arguments, result, sdk) } ?: result.error(...)

Reply:
The native SDK methods expects a non null value as its parameter. But the call.arguments can be a null value. As of such I use !!. If we want we can have a null check before this method, but then we wont able to pass null values to the SDK

Reply:
setDeviceName and setDeviceMetadata should not accept null values

Reply:
@AshishSynoriq regardless of whether or not the flutter method does not allow for null values, the bottom line is that if `call.arguments` is nullable (which it is) we should check each method call using `call.arguments` with `?.let { if non null } ?: result.error(...)` as @pavel-kuznetsov-hypertrack suggests.

Reply:
Then the current wrapper handles this

Reply:
What do you mean?

Reply:
i mean, do not use `!!`